docs/deploying a sample wildfly application.md

```diff
@@ -0,0 +1,144 @@
+# Deploying a sample WildFly application
+
+User Story:
+User downloads and starts Minishift (which deploys an instance of OpenShift Origin locally) in order to launch a sample Hello World application using WildFly.
+
+<!-- MarkdownTOC -->
+
+- [Objective](#objective)
+- [Prerequisites](#prerequisites)
+- [Deploying a Hello World WildFly application](#deploying-a-hello-world-wildfly-application)
+- [Additional resources](#additional resources)
+
+
+<!-- /MarkdownTOC -->
+
+<a name="objective"></a>
+## Objective
+To deploy a Hello World application using WildFly, on an OpenShift Origin instance provisioned by Minishift.
+
+<a name="prerequisites"></a>
+## Prerequisites:
+You can create a new application in OpenShift Origin either from a source code, an image or a template. In this example we use source code along with the WildFly Imagestream to deploy a Hello World application.
+
+1. To use source code to deploy the application, you need to have your source code hosted on the internet and accessible through GitHub. Fork and clone this [Sample Application](https://github.com/openshiftdemos/os-sample-java-web) repository. The Sample Application used here contains the source code for a simple Java web application with a single Jsp file that prints out a Hello World message and is hosted on GitHub.  
```

Comment:
"you need to have your source code hosted on the internet and accessible through GitHub" - that depends, if you want to modify it you need your own fork you can push to, but if you just want to get the demo working you can point to the actual demo repo as well. The various app templates usually per default use the main example repo, which you can build directly as well. It's jsut you won't be able to make modifications to the code.